lib/butter.js

```diff
@@ -1,81 +1,114 @@
-
-  // 3000ms timeout by default
-  var timeout = timeout || 3000;
-
-  var requestMethods = this._prepMethods(apiToken, testMode, timeout, axiosHook);
-
-  this._prepResources(requestMethods);
+import { BUTTER_CLIENT_VERSION } from './constants.js';
+
+import Author from './resources/Author.js';
+import Category from './resources/Category.js';
+import Content from './resources/Content.js';
+import Feed from './resources/Feed.js';
+import Page from './resources/Page.js';
+import Post from './resources/Post.js';
+import Tag from './resources/Tag.js';
+
+
+/**
+ * An object mapping the names of resources to their corresponding modules.
+ * Each module provides methods for interacting with a specific type of resource in the ButterCMS API.
+ * 
+ * @type {Object}
+ * @property {Object} Author - The module for interacting with author resources.
+ * @property {Object} Category - The module for interacting with category resources.
+ * @property {Object} Content - The module for interacting with content resources.
+ * @property {Object} Feed - The module for interacting with feed resources.
+ * @property {Object} Page - The module for interacting with page resources.
+ * @property {Object} Post - The module for interacting with post resources.
+ * @property {Object} Tag - The module for interacting with tag resources.
+ */
+const resources = {
+    Author,
+    Category,
+    Content,
+    Feed,
+    Page,
+    Post,
+    Tag,
 }
 
-Butter.prototype = {
-  version: BUTTER_CLIENT_VERSION,
-  _prepResources: function(requestMethods) {
-    for (var name in resources) {
-      this[
-        name[0].toLowerCase() + name.substring(1)
-      ] = new resources[name](requestMethods);
+/**
+ * Default export function for the ButterCMS library.
+ * 
+ * @param {string} apiToken - The API token for the ButterCMS service.
+ * @param {Object} [options={}] - Optional configuration options for the ButterCMS service.
+ * @throws Will throw an error if the API token is not set.
+ * @returns {Butter} An instance of the ButterCMS service.
+ */
+export default function (apiToken, options = {}) {
+    if (!apiToken) {
+        throw 'ButterCMS API token not set';
     }
-  },
-  _prepMethods: function(apiToken, testMode, timeout, axiosHook) {
-    return {
-      get: function(url, params) {
-        var headers = {'X-Butter-Client': 'JS/' + BUTTER_CLIENT_VERSION}
-        // detect if library is called in server side
-        if (typeof window === 'undefined'){
-          headers['Accept-Encoding'] = 'gzip'
-        }
-        var conn = axios.create({
-          baseURL: 'https://api.buttercms.com/v2',
-          timeout: timeout,
-          headers: headers
-        });
-
-        if (axiosHook) {
-          axiosHook(conn)
-        }
-
-        var params = params || {};
 
-        // Add api token to query string
-        params.auth_token = apiToken;
+    if (!(this instanceof Butter)) {
+        return new Butter(apiToken, options);
+    }
```

Comment:
Is this to make it singelton?

Reply:
I guess so, that was my take on it. This was something that was in the version prior to this that I decided not to mess with.